File: Blind 75/Best Time to Buy and Sell Stock.java
class Solution {
    public int maxProfit(int[] prices) {
        // Time O(n)
        // Because it iterates through the array once of n length

        // Space O(1)
        // Because the space stays the same regardless of prices length
        int lowest = Integer.MAX_VALUE;
        int maxProf = 0;
        for (int i = 0; i < prices.length; i++) {
            if (prices[i] < lowest) {
                lowest = prices[i];
            }
            maxProf = Math.max(maxProf, prices[i] - lowest);
        } 
        return maxProf;
    }
}

File: Python Blind 75/Arrays/Best Time to Buy and Sell Stock.py
class Solution:
    def maxProfit(self, prices: List[int]) -> int:
        # Time O(n), Space O(1)
        maxProf = 0
        minBuy = prices[0]

        for num in prices:
            maxProf = max(maxProf, num - minBuy)
            minBuy = min(minBuy, num)
        
        return maxProf